File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.swift
import Foundation

func solution(_ keyinput:[String], _ board:[Int]) -> [Int] {
    
    var result = [0, 0]
    
    keyinput.forEach {
        if $0 == "up" {
            if result[1] < (board[1] / 2) { result[1] += 1 }
        } else if $0 == "down" {
            if -(board[1] / 2) < result[1] { result[1] -= 1 }
        } else if $0 == "right" {
            if result[0] < (board[0] / 2) { result[0] += 1 }
        } else {
            if -(board[0] / 2) < result[0] { result[0] -= 1 }
        }
    }
    return result
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keyinput, vector<int> board) {
    vector<int> answer(2, 0);
    int width = board[0] / 2;
    int height = board[1] / 2;

    for(string s : keyinput) {
        if(s == "left" && -width < answer[0] ) {
            answer[0]--;
        } else if( s == "right" && width > answer[0]) {
            answer[0]++;
        } else if (s == "up" && height > answer[1]) {
            answer[1]++;
        } else if (s == "down" && -height < answer[1])  {
            answer[1]--;
        }
    }

    return answer;
}